specification/dartLangSpec.tex

```diff
@@ -16245,18 +16246,20 @@ \subsection{Assignable Expressions}
 \LMLabel{assignableExpressions}
 
 \LMHash{}%
-Assignable expressions are expressions
+\IndexCustom{Assignable expressions}{assignable expression} are terms
 that can appear on the left hand side of an assignment.
-This section describes how to evaluate these expressions
+This section describes how to evaluate these terms
 when they do not constitute the complete left hand side of an assignment.
+The semantics of an assignment as a whole is described elsewhere
+(\ref{assignment}).
 
-\rationale{%
-Of course, if assignable expressions
-always appeared \emph{as} the left hand side,
-one would have no need for their value,
-and the rules for evaluating them would be unnecessary.
-However, assignable expressions can be subexpressions of other expressions
-and therefore must be evaluated.%
+\commentary{%
+Assignable expressions can be subexpressions of other expressions,
+in which case they need to be evaluated.
+For example, if we are assigning a value to an instance variable \code{x}
+of a receiver $e$ by means of the expression \code{$e$.x\,\,=\,\,42},
+we need to evaluate $e$,
```

Comment:
... to an object,

But it's not actually clear why this is relevant, because we haven't said that `e` might be an assignable expression. This is just about expressions, not assignable expressions.

Reply:
It is about an assignable expression that contains an expression as the subterm `$e$`: We can't have `\code{$e$.x = 42}` unless `$e$.x` is an assignable expression.

Given that this is commentary, I'm jumping to the claim that `$e$` is an expression (that's often true, and that is indeed the case where expression evaluation is needed). Given that `$e$` could be arbitrarily complex, we certainly want to rely on expression evaluation (to avoid repeating every rule about expression evaluation here, just because the grammar rule doesn't parse `$e$` as an expression).